### DISCUSSION TITLE
how to backup docker container logs before be overwritten?

### DISCUSSION THREAD

**Discussion by FinalDes** [BretFisher/ama#197] [Q&A]
how to backup docker container logs before be overwritten?

**Comment by BretFisher** [comment]
Ideally, you don't back them up manually. Two main ways to manage docker container logs:

1. Set the log limits in the docker daemon config, or on each docker run, so you can set how big they can get and how long they are kept. Details at https://docs.docker.com
2. Use a docker log driver (see all the built-in ones with `docker info | grep Log`) on every container run or daemon config to send logs off to a 3rd party. Either your own hosted log storage or a cloud provider or SaaS.
